src/utils/Logging.ts

```diff
@@ -702,68 +702,71 @@ export default class Logging {
     if (!log.tenantID || log.tenantID === '') {
       log.tenantID = Constants.DEFAULT_TENANT;
     }
-    // Log
-    await LoggingStorage.saveLog(log.tenantID, log);
+
     // Log in Cloud Foundry
     if (Configuration.isCloudFoundry()) {
       // Bind to express app
       CFLog.logMessage(Logging.getCFLogLevel(log.level), log.message);
     }
+
+    // Log
+    return LoggingStorage.saveLog(log.tenantID, log);
   }
 
-  private static anonymizeSensitiveData(message: any): void {
+  private static anonymizeSensitiveData(message: any): any {
     if (!message || typeof message === 'number' || Utils.isBoolean(message) || typeof message === 'function') {
-      // eslint-disable-next-line no-useless-return
-      return;
-    } else if (typeof message === 'string') {
+      return message;
+    } else if (typeof message === 'string') { // If the message is a string
+      // Check if it is a query string
+      const dataParts: string[] = message.split('&');
+      if (dataParts.length > 1) {
+        for (let i = 0; i < dataParts.length; i++) {
+          const dataPart = dataParts[i];
+          for (const sensitiveData of Constants.SENSITIVE_DATA) {
+            if (dataPart.toLowerCase().startsWith(sensitiveData.toLocaleLowerCase())) {
+              // Anonymize each query string part
+              dataParts[i] = dataPart.substring(0, sensitiveData.length + 1) + Constants.ANONYMIZED_VALUE;
+            }
+          }
+        }
+        message = dataParts.join('&');
+        return message;
+      }
+      // Check if it is a string, but not a query string
       for (const sensitiveData of Constants.SENSITIVE_DATA) {
-        // Anonymize
-        message.replace(new RegExp(sensitiveData, 'gi'), Constants.ANONYMIZED_VALUE);
+        if (new RegExp(sensitiveData + '(\\s*)[:=,;](\\s*)(.*)', 'gi').test(message)) { // If it contains sensitive data and it matches the RegExp, Anonymize
+          message = message.substring(0, sensitiveData.length + 1) + Constants.ANONYMIZED_VALUE;
+          return message;
```

Comment:
If you have a string like:
'password=mypass+name=serge'
It'll end up like this if I'm not wrong (name is before password in the list of anon fields):
'password=mypass+name=####'
To avoid this, I would simply anonymize the whole string.
Let me know.

Reply:
The code should anonymize everything after password= and name=, the regexp is using 'gi' flag. 
But strings combination of several replacement at once shall be added to the unit tests.

Reply:
Then also fixed in the code above which returns the string after the first replacement I think

Reply:
The string you are mentioning ('password=mypass+name=serge') is indeed badly obfuscated at the moment. That is because '+' is not treated as a separator in strings (like '&' is). Should I treat it as one and implement that logic + add corresponding tests?

Reply:
Also - yes, the code you are mentioning is returning the string after the first replacement, because it does not contain '&' (is not a query string).

Reply:
I think that you just have to replace the code below:
```
if (new RegExp(sensitiveData + '(\\s*)[:=,;](\\s*)(.*)', 'gi').test(message)) { // If it contains sensitive data and it matches the RegExp, Anonymize
          message = message.substring(0, sensitiveData.length + 1) + Constants.ANONYMIZED_VALUE;
          return message;
```

by this one:

```
if (new RegExp(sensitiveData + '(\\s*)[:=,;](\\s*)(.*)', 'gi').test(message)) { // If it contains sensitive data and it matches the RegExp, Anonymize
          return Constants.ANONYMIZED_VALUE;
```

Reply:
Done + added specific test for checking query string obfuscation is done properly.